File: 0115/nm.c
#include <stdio.h>

int n, m;
int visit[9] = {0}, combi[9];

int addNumToArr(int index)
{
  if (index == m)
  {
    for (int i = 0; i < m; i++)
    {
      printf("%d ", combi[i]);
    }
    printf("\n");
  }

  for (int i = 0; i < n; i++)
  {
    if (!visit[i])
    {
      combi[index] = i + 1;
      visit[i] = 1;
      addNumToArr(index + 1);
      visit[i] = 0;
    }
  }
}

int main(void)
{
  scanf("%d %d", &n, &m);
  addNumToArr(0);

  return 0;
}

File: 0115/nm.py
def addNum(index):
  global n
  global visit
  global combination

  if index == m:
    print(" ".join(map(str, combination)))
    return

  for i in range(n):
    if not visit[i]:
      if index == 0 or combination[index-1] < i+1:
        combination[index] = i+1
        visit[i] = True
        addNum(index + 1)
        visit[i] = False

n, m = map(int, input().split())
visit = [False for _ in range(n)]
combination = [0 for _ in range(m)]
addNum(0)

